src/expressions/await-expr.md

```diff
@@ -0,0 +1,50 @@
+# Await expressions
+
+> **<sup>Syntax</sup>**\
+> _AwaitExpression_ :\
+> &nbsp;&nbsp; [_Expression_] `.` `await`
+
+Await expressions are legal only within `async` contexts, like an
+`async fn` or an `async` block. They operate on a future. Their effect
+is to "pause" the current computation until the given future is ready
+to produce a value.
+
+More specifically, a `f.await` expression has the following effect.
+
+1. evaluate `f` to a future `tmp` using the `IntoFuture:into_future` method;
```

Comment:
```suggestion
1. evaluate `f` to a future temporary variable `tmp` using the `IntoFuture:into_future` method;
```

I don't like "future" there, but don't really see a better way to succinctly convey the information. It should probably link to the Future trait as well.